src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -239,7 +239,8 @@ CHIP_ERROR AES_CCM_decrypt(const uint8_t * ciphertext, size_t ciphertext_length,
 
     // Pass in expected tag
     VerifyOrExit(CanCastTo<int>(tag_length), error = CHIP_ERROR_INVALID_ARGUMENT);
-    result = EVP_CIPHER_CTX_ctrl(context, EVP_CTRL_CCM_SET_TAG, static_cast<int>(tag_length), (void *) tag);
+    result = EVP_CIPHER_CTX_ctrl(context, EVP_CTRL_CCM_SET_TAG, static_cast<int>(tag_length),
+                                 const_cast<void *>(reinterpret_cast<const void *>(tag)));
```

Comment:
Isn't static_cast enough to convert tag to "const void*"? And in other places below where you convert PTR* to/from void*?

Reply:
EVP_CIPHER_CTX_ctrl seems to be a heavily overloaded C function - it's used for writes and reads. So they have to choose one signature or another.

Reply:
Yes, static_cast is enough to convert for void *

(although maybe it shouldn't be, since this allows achieving the same effect as reinterpret_cast using 2 static_casts)